IdentityCore/src/validation/MSIDAADAuthority.m

```
@@ -0,0 +1,247 @@
+
+@end
+
+@implementation MSIDAADAuthority
+
+@synthesize openIdConfigurationEndpoint;
+
+- (instancetype)initWithURL:(NSURL *)url
+                    context:(id<MSIDRequestContext>)context
+                      error:(NSError **)error
+{
+    self = [super initWithURL:url context:context error:error];
+    if (self)
+    {
+        _url = [self.class normalizedAuthorityUrl:url context:context error:error];
+        if (!_url) return nil;
+        _tenant = [self.class tenantFromAuthorityUrl:self.url context:context error:error];
+        _authorityCache = [MSIDAadAuthorityCache sharedInstance];
+        _authorityFactory = [MSIDAuthorityFactory new];
+    }
+    
+    return self;
+}
+
+- (nullable instancetype)initWithURL:(nonnull NSURL *)url
+                           rawTenant:(NSString *)rawTenant
+                             context:(nullable id<MSIDRequestContext>)context
+                               error:(NSError **)error
+{
+    self = [self initWithURL:url context:context error:error];
+    if (self)
+    {
+        if (rawTenant && self.tenant.type != MSIDAADTenantTypeIdentifier)
+        {
+            _url = [NSURL URLWithString:[NSString stringWithFormat:@"https://%@/%@", [_url msidHostWithPortIfNecessary], rawTenant]];
+            
+            if (![self.class isAuthorityFormatValid:_url context:context error:error]) return nil;
+            
+            _tenant = [self.class tenantFromAuthorityUrl:self.url context:context error:error];
+        }
+    }
+    
+    return self;
+}
+
+- (NSURL *)networkUrlWithContext:(id<MSIDRequestContext>)context
+{
+    return [self.authorityCache networkUrlForAuthority:self context:context];
+}
+
+- (NSURL *)cacheUrlWithContext:(id<MSIDRequestContext>)context
+{
+    __auto_type universalAuthorityURL = [self universalAuthorityURL];
+    __auto_type authority = (MSIDAADAuthority *)[self.authorityFactory authorityFromUrl:universalAuthorityURL context:context error:nil];
```

nit: why are we ignoring error?